### ISSUE TITLE
ValueError: RSA key format is not supported

### ISSUE THREAD

**Issue by balleno75** [xhlove/WVClient3#12]
Hi, I have a problem extracting the key from a disneyplus content. I indicate the server address and the pssh and it generates that RSA key format is not supported

python .\wvclient.py --license-url "https://disney.playback.edge.bamgrid.com/widevine/v1/obtain-license" --pssh "AAAAMnBzc2gAAAAA7e+LqXnWSs6jyCfc1R0h7QAAABISEPV9OJvzPU++nZ6fz+1mU5I=" --headers "customdata:PEtleU9TQXV0aGVudGljYXRpb25YTUw+PERhdGE+PEdlbmVyYXRpb25UaW1lPjIwMTYtMTEtMTkgMDk6MzQ6MDEuOTkyPC9HZW5lcmF0aW9uVGltZT48RXhwaXJhdGlvblRpbWU+MjAyNi0xMS0xOSAwOTozNDowMS45OTI8L0V4cGlyYXRpb25UaW1lPjxVbmlxdWVJZD4wZmZmMTk3YWQzMzQ0ZTMyOWU0MTA0OTIwMmQ5M2VlYzwvVW5pcXVlSWQ+PFJTQVB1YktleUlkPjdlMTE0MDBjN2RjY2QyOWQwMTc0YzY3NDM5N2Q5OWRkPC9SU0FQdWJLZXlJZD48V2lkZXZpbmVQb2xpY3kgZmxfQ2FuUGxheT0idHJ1ZSIgZmxfQ2FuUGVyc2lzdD0iZmFsc2UiIC8+PFdpZGV2aW5lQ29udGVudEtleVNwZWMgVHJhY2tUeXBlPSJIRCI+PFNlY3VyaXR5TGV2ZWw+MTwvU2VjdXJpdHlMZXZlbD48L1dpZGV2aW5lQ29udGVudEtleVNwZWM+PEZhaXJQbGF5UG9saWN5IC8+PExpY2Vuc2UgdHlwZT0ic2ltcGxlIiAvPjwvRGF0YT48U2lnbmF0dXJlPk1sNnhkcU5xc1VNalNuMDdicU8wME15bHhVZUZpeERXSHB5WjhLWElBYlAwOE9nN3dnRUFvMTlYK1c3MDJOdytRdmEzNFR0eDQydTlDUlJPU1NnREQzZTM4aXE1RHREcW9HelcwS2w2a0JLTWxHejhZZGRZOWhNWmpPTGJkNFVkRnJUbmxxU21raC9CWnNjSFljSmdaUm5DcUZIbGI1Y0p0cDU1QjN4QmtxMUREZUEydnJUNEVVcVJiM3YyV1NueUhGeVZqWDhCR3o0ZWFwZmVFeDlxSitKbWI3dUt3VjNqVXN2Y0Fab1ozSHh4QzU3WTlySzRqdk9Wc1I0QUd6UDlCc3pYSXhKd1ZSZEk3RXRoMjhZNXVEQUVZVi9hZXRxdWZiSXIrNVZOaE9yQ2JIVjhrR2praDhHRE43dC9nYWh6OWhVeUdOaXRqY2NCekJvZHRnaXdSUT09PC9TaWduYXR1cmU+PC9LZXlPU0F1dGhlbnRpY2F0aW9uWE1MPg=="
failed to parse cert as SignedMessage
Traceback (most recent call last):
  File ".\wvclient.py", line 154, in <module>
    main(args)
  File ".\wvclient.py", line 133, in main
    challenge = wvdecrypt.get_challenge()
  File "C:\Users\*\Downloads\WVClient3-main\pywidevine\decrypt\wvdecrypt.py", line 42, in get_challenge
    return self.cdm.get_license_request(self.session)
  File "C:\Users\*\Downloads\WVClient3-main\pywidevine\cdm\cdm.py", line 211, in get_license_request
    service_public_key = RSA.importKey(session.service_certificate._DeviceCertificate.PublicKey)
  File "C:\Users\*\AppData\Local\Programs\Python\Python38\lib\site-packages\Crypto\PublicKey\RSA.py", line 788, in import_key
    raise ValueError("RSA key format is not supported")
ValueError: RSA key format is not supported

The same error is generated, with the file downloaded from Netflix,

python.exe .\wvclient.py --init-path "C:\Users\*\Downloads\descarga.mp4" --license-url "https://www.netflix.com/nq/msl_v1/cadmium/pbo_licenses/%5E1.0.0/router" --headers "customdata:PEtleU9TQXV0aGVudGljYXRpb25YTUw+PERhdGE+PEdlbmVyYXRpb25UaW1lPjIwMTYtMTEtMTkgMDk6MzQ6MDEuOTkyPC9HZW5lcmF0aW9uVGltZT48RXhwaXJhdGlvblRpbWU+MjAyNi0xMS0xOSAwOTozNDowMS45OTI8L0V4cGlyYXRpb25UaW1lPjxVbmlxdWVJZD4wZmZmMTk3YWQzMzQ0ZTMyOWU0MTA0OTIwMmQ5M2VlYzwvVW5pcXVlSWQ+PFJTQVB1YktleUlkPjdlMTE0MDBjN2RjY2QyOWQwMTc0YzY3NDM5N2Q5OWRkPC9SU0FQdWJLZXlJZD48V2lkZXZpbmVQb2xpY3kgZmxfQ2FuUGxheT0idHJ1ZSIgZmxfQ2FuUGVyc2lzdD0iZmFsc2UiIC8+PFdpZGV2aW5lQ29udGVudEtleVNwZWMgVHJhY2tUeXBlPSJIRCI+PFNlY3VyaXR5TGV2ZWw+MTwvU2VjdXJpdHlMZXZlbD48L1dpZGV2aW5lQ29udGVudEtleVNwZWM+PEZhaXJQbGF5UG9saWN5IC8+PExpY2Vuc2UgdHlwZT0ic2ltcGxlIiAvPjwvRGF0YT48U2lnbmF0dXJlPk1sNnhkcU5xc1VNalNuMDdicU8wME15bHhVZUZpeERXSHB5WjhLWElBYlAwOE9nN3dnRUFvMTlYK1c3MDJOdytRdmEzNFR0eDQydTlDUlJPU1NnREQzZTM4aXE1RHREcW9HelcwS2w2a0JLTWxHejhZZGRZOWhNWmpPTGJkNFVkRnJUbmxxU21raC9CWnNjSFljSmdaUm5DcUZIbGI1Y0p0cDU1QjN4QmtxMUREZUEydnJUNEVVcVJiM3YyV1NueUhGeVZqWDhCR3o0ZWFwZmVFeDlxSitKbWI3dUt3VjNqVXN2Y0Fab1ozSHh4QzU3WTlySzRqdk9Wc1I0QUd6UDlCc3pYSXhKd1ZSZEk3RXRoMjhZNXVEQUVZVi9hZXRxdWZiSXIrNVZOaE9yQ2JIVjhrR2praDhHRE43dC9nYWh6OWhVeUdOaXRqY2NCekJvZHRnaXdSUT09PC9TaWduYXR1cmU+PC9LZXlPU0F1dGhlbnRpY2F0aW9uWE1MPg=="
failed to parse cert as SignedMessage
Traceback (most recent call last):
  File ".\wvclient.py", line 154, in <module>
    main(args)
  File ".\wvclient.py", line 133, in main
    challenge = wvdecrypt.get_challenge()
  File "C:\Users\*\Downloads\WVClient3-main\pywidevine\decrypt\wvdecrypt.py", line 42, in get_challenge
    return self.cdm.get_license_request(self.session)
  File "C:\Users\*\Downloads\WVClient3-main\pywidevine\cdm\cdm.py", line 211, in get_license_request
    service_public_key = RSA.importKey(session.service_certificate._DeviceCertificate.PublicKey)
  File "C:\Users\*\AppData\Local\Programs\Python\Python38\lib\site-packages\Crypto\PublicKey\RSA.py", line 788, in import_key
    raise ValueError("RSA key format is not supported")
ValueError: RSA key format is not supported

Please can help?. I don't know if the problem is due to the cdmapi library, although for filmin.es, the key is extracted correctly, of course, indicating the MPD address.

Thanks. A greeting.

**Comment by quaternoon** [comment]
Hi,

Very great tool. I got it working (can run the example) but I got the same error as above when trying to obtain the key for this [show](https://www.canalplus.com/jeunesse/les-cahiers-d-esther/h/14804871_50001). I thought it was a mistake of mine (not sending the correct header) but maybe it is not. In my case the string, the string sent to the RSA function is empty.

Any help will be appreciated. Thanks.
